### PR TITLE
Added field parameter to the from_json method with other required cha…

### LINKED ISSUE #576: Expose the `field` parameter for JSON data sources in Seq2Seq tasks
## 🚀 Feature

HF gives a `field` parameter that can be used with JSON data. We should expose it through the `from_json` mehod

### PR DESCRIPTION
**Pull request by karthikrangasai**
…nges.

## What does this PR do?

Fixes #576 

## Before submitting
- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/lightning-flash/tree/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests? [not needed for typos/docs]
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/lightning-flash/blob/master/CHANGELOG.md)?

## PR review
 - [x] Is this pull request ready for review? (if not, please submit in draft mode)

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: flash/core/data/data_module.py [hunk 1]
```diff
@@ -839,6 +839,7 @@ def from_json(
         batch_size: int = 4,
         num_workers: Optional[int] = None,
         sampler: Optional[Sampler] = None,
+        field: str = None,
```
Comment (ethanwharris) [inline flash/core/data/data_module.py:842]:
```suggestion
        field: Optional[str] = None,
```

**Review by ethanwharris — APPROVED** [review]
Awesome, LGTM :smiley: Mind also updating `CHANGELOG.md`?

**Comment by karthikrangasai** [comment]
I have updated the PR with both the changes.

File: flash/core/data/data_module.py [hunk 2]
```diff
@@ -936,13 +938,35 @@ def from_json(
...
+            # In the case where the data is of the form:
+            # {
+            #     "version": 0.0.x,
+            #     "data": {
+            #         {
+            #             "input_field" : "input_data",
+            #             "target_field" : "target_output"
+            #         },
+            #         ...
+            #     }
```
Comment (ethanwharris) [inline flash/core/data/data_module.py:951]:
```suggestion
            #     "data": [
            #         {
            #             "input_field" : "input_data",
            #             "target_field" : "target_output"
            #         },
            #         ...
            #     ]
```

**Pull request merged**